### PR TITLE
Increase the rate of topic creation

### PR DESCRIPTION
**Pull request by suiyuzeng**
### Motivation

*The count of topic created in 1 second is less than 200, when the topic count of the namespace is about 200,000..*

### Modifications

*The reason is that the time cost of topic exist checking is too long when the topic count is large.
In org.apache.pulsar.broker.admin.AdminResource#checkTopicExistsAsync, all the topics would be checked one by one. As the count of topic grows, the time cost will become longer.
Can we use org.apache.pulsar.broker.namespace.NamespaceService#checkTopicExists to check if the topic is exits?*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: ( no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: ( no )

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  code optimize
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by codelipenghui — CHANGES_REQUESTED** [review]
Nice catch! But I think this will break the current behavior. Currently, we do not allow the duplicate topic name with the persistent or non-persistent domain, with this change it will only check the persistent topic duplication.

I think we should add the non-persistent topic duplication check back.

**Comment by suiyuzeng** [comment]
> Nice catch! But I think this will break the current behavior. Currently, we do not allow the duplicate topic name with the persistent or non-persistent domain, with this change it will only check the persistent topic duplication.
> 
> I think we should add the non-persistent topic duplication check back.

Got it. I will update later.

**Comment by suiyuzeng** [comment]
@codelipenghui hi, can you please take a look?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java [hunk 1]
```diff
@@ -742,19 +742,28 @@ protected void internalCreatePartitionedTopic(AsyncResponse asyncResponse, int n
...
+        if (topicName.isPersistent()) {
+            String path = ZkAdminPaths.partitionedTopicPath(topicName);
+            return pulsar().getPulsarResources().getNamespaceResources().getPartitionedTopicResources().
+                    existsAsync(path).thenCombine(pulsar().getLocalZkCacheService().
+                            managedLedgerExists(topicName.getPersistenceNamingEncoding()),
+                    (isPartitionedExists, isNonPartitionedExists) -> isPartitionedExists || isNonPartitionedExists);
+
+        } else {
+            return pulsar().getNamespaceService().getListOfTopics(topicName.getNamespaceObject(),
+                    CommandGetTopicsOfNamespace.Mode.NON_PERSISTENT)
```
Comment (315157973) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:754]: We have multiLayerTopicsMap in BrokerService, we don't need to traverse all NON_PERSISTENT, and then traverse again to determine whether it exists.
We only need to traverse the bundles, and then directly map.contains can do it, usually the number of bundles is not too much

**Review by eolivelli — COMMENTED** [review]
very nice catch !
I am happy to see an improvement here

do we have specific test cases around this function ?

**Comment by suiyuzeng** [comment]
@315157973 @eolivelli thanks，i will Optimize the code and add some test about this function.

**Comment by 315157973** [comment]
@suiyuzeng Hello, please go on

**Comment by suiyuzeng** [comment]
> @suiyuzeng Hello, please go on

Sorry, a little busy recently~
Find another problem, such as the test case "testCreatePartitionedTopicHavingNonPartitionTopicWithPartitionSuffix".  Some non partition topic has the partition suffix. In this case, we can not use the cache to check anymore. There is some checking about this issue in the current code, to ensure that we can not create no no-parititon topic whit -partition-xxx suffix.  How about add a config? If a cluster has the no-partition whit the suffix, use the current way to check duplication. If no no-paritition topic whit the suffix, use the cache to check.

**Comment by hangc0276** [comment]
> > @suiyuzeng Hello, please go on
> 
> Sorry, a little busy recently~
> Find another problem, such as the test case "testCreatePartitionedTopicHavingNonPartitionTopicWithPartitionSuffix". Some non partition topic has the partition suffix. In this case, we can not use the cache to check anymore. There is some checking about this issue in the current code, to ensure that we can not create no no-parititon topic whit -partition-xxx suffix. How about add a config? If a cluster has the no-partition whit the suffix, use the current way to check duplication. If no no-paritition topic whit the suffix, use the cache to check.

@suiyuzeng  Nice catch!
1. I am cutting 2.8.1, would you have time to address @315157973 's comments? 
2. for the restriction for reserved key-words for topic name, such as -partition-xx suffix for non-partitioned topic name, we have started a discussion for this issue: https://lists.apache.org/list.html?dev@pulsar.apache.org

**Comment by hangc0276** [comment]
move to 2.8.2

**Comment by 315157973** [comment]
move to 2.8.3

**Comment by michaeljmarshall** [comment]
Removing the `release/2.8.3` label as this change will target master and probably won't be cherry picked back to stable branches because it is an optimization.

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

I believe this patch is worth continuing, but holding a stale patch doesn't work. If @suiyuzeng can rebase and resubmit the patch, I'm glad to follow the PR lifecycle. Otherwise, I may pick it up later. Close.

**Pull request closed without merging**